analyzers/src/SonarAnalyzer.Common/Rules/ParameterAssignedToBase.cs

```diff
@@ -38,38 +38,27 @@ public abstract class ParameterAssignedToBase<TSyntaxKind, TIdentifierNameSyntax
 
         protected ParameterAssignedToBase() : base(DiagnosticId) { }
 
-        protected sealed override void Initialize(SonarAnalysisContext context)
-        {
+        protected sealed override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSyntaxNodeActionInNonGenerated(
-                Language.GeneratedCodeRecognizer,
-                c =>
+                Language.GeneratedCodeRecognizer, c =>
                 {
-                    var left = Language.Syntax.AssignmentLeft(c.Node);
-                    if (c.SemanticModel.GetSymbolInfo(left).Symbol is { } symbol
-                        && (symbol is IParameterSymbol { RefKind: RefKind.None } || IsAssignmentToCatchVariable(symbol, left))
-                        && (!IsReadBefore(c.SemanticModel, symbol, c.Node)))
+                    foreach (var target in Language.Syntax.AssignmentTargets(c.Node))
                     {
-                        c.ReportIssue(Diagnostic.Create(SupportedDiagnostics[0], left.GetLocation(), left.ToString()));
+                        if (c.SemanticModel.GetSymbolInfo(target).Symbol is { } symbol
+                            && (symbol is IParameterSymbol { RefKind: RefKind.None } || IsAssignmentToCatchVariable(symbol, target))
+                            && !IsReadBefore(c.SemanticModel, symbol, c.Node))
+                        {
+                            c.ReportIssue(Diagnostic.Create(SupportedDiagnostics[0], target.GetLocation(), target.ToString()));
+                        }
                     }
                 },
                 Language.SyntaxKind.SimpleAssignment);
-        }
 
-        private bool IsReadBefore(SemanticModel semanticModel, ISymbol parameterSymbol, SyntaxNode assignment)
-        {
-            // Same problem as in VB.NET / IsAssignmentToCatchVariable:
-            // parameterSymbol.DeclaringSyntaxReferences is empty for Catch syntax in VB.NET as well as for indexer syntax for C#
-            // https://github.com/dotnet/roslyn/issues/6209
-            var stopLocation = parameterSymbol.Locations.FirstOrDefault();
-            if (stopLocation == null)
-            {
-                return true; // If we can't find the location, it's going to be FN
-            }
```

Comment:
Code coverage has shown, that this code path is not hit anymore and so I removed it. The bug is fixed in newer Roslyn versions. I can't tell if it is really safe to remove the check.

Reply:
@pavel-mikula-sonarsource  Maybe this should be kept for backward compatibility?

Reply:
We need to keep it as long as we support the old version, unfortunately.